File: Java/src/com/company/Lists/LC24_swapPairs.java
package com.company.Lists;
/*
    LC24
    Linkedlist
 */
public class LC24_swapPairs {
    public ListNode swapPairs(ListNode head) {
        // corner case
        if(head == null) return null;
        if(head.next == null) return head;
        ListNode dummyhead = new ListNode(-1);
        dummyhead.next = head;
        ListNode curNode = dummyhead;
        while(curNode.next != null && curNode.next.next != null ){
            ListNode temp = curNode;
            ListNode nextNode = temp.next;

            temp.next = nextNode.next;
            nextNode.next = nextNode.next.next;
            temp.next.next = nextNode;

            curNode = curNode.next.next;
        }
        return dummyhead.next;
    }
    public ListNode swapPairsRec(ListNode head){
                /*
            recursively
        */
        // corner case
        if(head == null || head.next == null) return head;
        ListNode temp = head.next;
        head.next = swapPairsRec(head.next.next);
        temp.next = head;
        return temp;
    }
}

File: python/List/LC24_swapPairs.py
def swapPairs(self, head: ListNode) -> ListNode:
        # dummy = ListNode(-1)
        # dummy.next = head
        # cur = dummy
        # while(cur and cur.next):
        #     if not cur.next.next:
        #         return dummy.next
        #     a = cur.next
        #     b = a.next
        #     nx = b.next
        #     cur.next = b
        #     b.next = a
        #     a.next = nx
        #     cur = a
        # return dummy.next
        if not head or not head.next:
            return head
        nextNode = self.swapPairs(head.next.next)
        tmp = head.next
        tmp.next = head
        head.next = nextNode
        return tmp
